modules/account-assignment/main.tf

```
@@ -2,7 +2,7 @@ resource "aws_ssoadmin_account_assignment" "this" {
   instance_arn       = local.sso_instance_arn
   permission_set_arn = var.account_assignment.permission_set_arn != null ? var.account_assignment.permission_set_arn : data.aws_ssoadmin_permission_set.this[0].arn
 
-  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this[0].id : data.aws_identitystore_user.this[0].id
+  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this.alternate_identifier : data.aws_identitystore_user.this.alternate_identifier
```

I feel like this shouldn't change? What was the rationale for this change?

```suggestion
  principal_id   = var.account_assignment.principal_type == "GROUP" ? data.aws_identitystore_group.this[0].id : data.aws_identitystore_user.this[0].id
```